scripts/ontology_to_ontology_mapping_tool/ontology_to_ontology_mapping_tool.py

```diff
@@ -0,0 +1,171 @@
+    if ambiguous_codes:
+        log_file.write("<hr><h2 id=\"one_to_many_mapping\">The following source codes mapped to multiple target codes:</h2>\n")
+        for code in ambiguous_codes:
+            log_file.write("<p><b>Source Code</b>: %s<br>\n" % (code))
+            log_file.write("<b>Target Codes</b>: %s<br>\n" % ambiguous_codes[code])
+    if unrecognized_codes:
+        log_file.write("<hr><h2 id=\"no_match\">The following source codes have no mapping available:</h2>\n")
+        for code in unrecognized_codes:
+            log_file.write("<p><b>Source Code: %s</b><br>\n" % ("&lt;blank&gt;" if len(code) == 0 else code))
+            log_file.write("<b>Mapping not available for the code\n")
+    if many_to_one_codes:
+        log_file.write("<hr><h2 id=\"many_to_one_mapping\">The following target codes mapped to multiple source codes:</h2>\n")
+        for code in many_to_one_codes:
+            log_file.write("<p><b>Target Code</b>: %s<br>\n" % (code))
+            log_file.write("<b>Source Codes</b>: %s<br>\n" % many_to_one_codes[code])
+
+    log_file.close()
+
+#Check if any of the ontology code headers are present in the clinical file
+def extract_source_code_validate_arguments(source_file,target_code):
+    #1. Check if the target_code input by the user is valid.
+    if target_code not in ACCEPTED_ONTOLOGIES:
+        print("Invalid target code: \'%s\'. \nThe list of acceptable codes are:" % (target_code))
+        for code in ACCEPTED_ONTOLOGIES:
+            print(code)
+        sys.exit(1)
+
+    #2. Check if the user input file has a valid source ontology header. If yes return the source code.
+    #If the source file has multiple ontologies, ask the user to enter the ontology to map on.
+    matched_headers = []
+    for header in source_file.columns:
+        if header.upper() in ACCEPTED_ONTOLOGIES and header.upper() not in matched_headers:
+            source_file.rename(columns={header:header.upper()}, inplace=True)
+            matched_headers.append(header.upper())
+    if len(matched_headers) == 1:
+        source_code = matched_headers[0]
+        print("The input file has the following ontology code: %s\n" % (source_code))
+    elif len(matched_headers) == 0:
+        print("\nThe input file does not contain any ontology code columns or the headers do not match to accepted values. Please check. \n\nThe list of acceptable ontology headers are:")
+        for code in ACCEPTED_ONTOLOGIES:
+            print(code)
+        print('\n')
+        sys.exit(1)
+    else:
+        print("The input file has the following ontology code columns: ")
+        for code in matched_headers:
+            print(code)
+        source_code = input("Please specify the source code column to map by: ")
+        if source_code not in ACCEPTED_ONTOLOGIES:
+            print('The Source code entered is invalid.')
+            sys.exit(1)
+
+    #3. Check if source_code and target_code are not the same
+    if source_code == target_code:
```

Comment:
good check --- maybe another good check to add would be to see if the target ontology column already exists in the source file. The user might have a source of oncotree and a target of ICD-O ... but what would happen if the ICD-O column is already present in their source file? I suppose it would be overwritten? Maybe that is ok.

Reply:
If the source file already has the columns, ICD-O in your example, the common columns in the source file will get the suffix `_source_file` ICDO_source_file here.

Reply:
ok, that seems reasonable. I guess if the user starts with ICDO column and then runs the script twice they may wind up with ICDO_source_file_source_file ... but we can leave it to them to clean up those columns I guess.